### ISSUE TITLE
[CSS] Nesting at-rules (eg. `@media`) breaks syntax highlighting

### ISSUE THREAD

**Issue by ryami333** [sublimehq/Packages#3946]
### What happened?

[CSS Nesting](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_nesting) supports nesting of at-rules (like `@media`, `@container` et al.) but currently the syntax highlighting for this is broken. You can create a minimum reproducible case by copy+pasting the examples from the MDN docs:

```css
.foo {
  display: grid;
  @media (orientation: landscape) {
    grid-auto-flow: column;
  }
}
```

Note that the "expanded nested CSS" syntax _does_ already highlight correctly. Eg.

```css
.foo {
  display: grid;
  @media (orientation: landscape) {
    & {
      grid-auto-flow: column;
    }
  }
}
```

Preview:
<img width="501" alt="Screenshot 2024-03-13 at 10 41 47" src="https://github.com/sublimehq/Packages/assets/12481532/edaa3cf0-5c5a-4672-8538-bb815d43df97">

This is going to be a focus area of [Interop 2024](https://web.dev/blog/interop-2024#nesting), which means that by the end of 2024 this is probably going to have wide uptake within the front-end development community. In fact, it's already [supported](https://caniuse.com/css-nesting) by most modern browsers.

**Comment by deathaxe** [comment]
This issue has already been fixed by #3898.

**Comment by michaelblyons** [comment]
Good bug report format, but duplicate of #3897.
